Add `rustls_ring` feature flag to support WSS on native

c-git opened the pull request:
- Provides a feature flag to enable ring to serve as a default crypto provider
 - Ensures that tls support is still provided even if tokio is enabled. Closes: #57

c-git committed:
feat: add support WSS

emilk commented on example_app/src/main.rs:
```diff
@@ -20,6 +20,12 @@ fn main() -> eframe::Result<()> {
 fn main_impl() -> Result<(), eframe::Error> {
     env_logger::init(); // Log to stderr (if you run with `RUST_LOG=debug`).
 
+    // Setup a CryptoProvider to be able to use wss
+    match rustls::crypto::ring::default_provider().install_default() {
+        Ok(()) => {} // Do nothing crypto provider install successful
+        Err(_) => log::warn!("failed to install CryptoProvider"),
+    }
```
Is this strictly required to connect over `wss:/`? If so, we should maybe add it to `ewebsock` proper (perhaps opt-in), or at least document it

c-git replied:
It is strictly required to have a provider to connect to WSS. That's a good idea. It would need to be feature flagged so users can disable it if they want to user a different provider. I can do a PR for that if you think that's a good idea.

c-git replied:
The problem only affects the native client. It works fine in the browser.  

![Screenshot of error](https://github.com/user-attachments/assets/c5f8b1a9-8801-4963-aff4-1ab7bb046a91)

Must be done exactly once. Will try to document it and add a optional function the users can call.

![Screenshot from docs](https://github.com/user-attachments/assets/b3b8b644-867a-45c1-a19b-13df6cfc1928)

c-git commented:
I'm was looking over my comment and found the error message looked wrong. It was wrong. I used https instead of wss.

c-git committed:
feat: make rustls/ring an opt in dep of ewebsock

emilk commented on ewebsock/Cargo.toml:
```diff
@@ -23,6 +23,10 @@ workspace = true
 [features]
 default = []
 
+## Enables the ring feature on the rustls create to allow connections from native apps to WSS (Not needed if your supply your own CryptoProvider)
```
```suggestion
## Enables the ring feature on the rustls create to allow connections from native apps to WSS.
## (Not needed if you supply your own CryptoProvider)
```

emilk committed:
fix small typo

c-git commented:
Given the recent news related to ring (RUSTSEC-2025-0007). I think it makes sense to change the feature to use a different provider.

c-git commented:
Will try to update once I get a chance but can't find how to set the pr to draft on the mobile client in the meantime.

c-git commented:
The advisory has been [withdrawn](https://rustsec.org/advisories/RUSTSEC-2025-0007.html) and others [have stepped](https://github.com/briansmith/ring/discussions/2414#discussioncomment-12391594) up to help.

c-git committed:
fix: support tls on native even with tokio

c-git commented:
The following extract from `cargo tree` shows the duplicate versions causing CI to fail for cargo-deny

```
echo_server v0.1.0 (/workspaces/ewebsock/echo_server)
└── tungstenite v0.24.0
    └── rustls-native-certs v0.7.3

ewebsock v0.8.0 (/workspaces/ewebsock/ewebsock)
└── tokio-tungstenite v0.24.0
    └── rustls-native-certs v0.8.0
```

c-git committed:
chore: cargo update

c-git committed:
chore: remove lint with error SHA1

Was also removed in egui at
https://github.com/emilk/egui/commit/4c1f344ef8d3ce091bde4fff61e984467a9635e6#diff-f3dbfb2563c3490394f44c26b51837018e79a79e75fd31e89b19e943a38317eaL62

Was giving a warning "warning: expected a type, found a static"

c-git committed:
chore: stop cargo-shear warning

Expected not to be used as it's added as a convince for end users

c-git commented:
When I run the `./cargo_deny.sh ` script I get an error for duplicate `getrandom`. I've done a `cargo update` but the older one wasn't able to be updated.  See output below.

```
+ cargo deny --all-features --log-level error --target aarch64-apple-darwin check
error[duplicate]: found 2 duplicate entries for crate 'getrandom'
   ┌─ /home/one/data/OpFiles/repos/GroupSideProjects/egui/ewebsock/Cargo.lock:81:1
   │  
81 │ ╭ getrandom 0.2.17 registry+https://github.com/rust-lang/crates.io-index
82 │ │ getrandom 0.3.4 registry+https://github.com/rust-lang/crates.io-index
   │ ╰─────────────────────────────────────────────────────────────────────┘ lock entries
   │  
   ├ getrandom v0.2.17
     └── ring v0.17.14
         ├── rustls v0.23.37
         │   ├── ewebsock v0.8.0
         │   │   └── example_app v0.1.0
         │   ├── tokio-rustls v0.26.4
         │   │   └── tokio-tungstenite v0.29.0
         │   │       └── ewebsock v0.8.0 (*)
         │   ├── tokio-tungstenite v0.29.0 (*)
         │   └── tungstenite v0.29.0
         │       ├── echo_server v0.1.0
         │       ├── ewebsock v0.8.0 (*)
         │       └── tokio-tungstenite v0.29.0 (*)
         └── rustls-webpki v0.103.10
             └── rustls v0.23.37 (*)
   ├ getrandom v0.3.4
     └── rand_core v0.9.5
         ├── rand v0.9.2
         │   └── tungstenite v0.29.0
         │       ├── echo_server v0.1.0
         │       ├── ewebsock v0.8.0
         │       │   └── example_app v0.1.0
         │       └── tokio-tungstenite v0.29.0
         │           └── ewebsock v0.8.0 (*)
         └── rand_chacha v0.9.0
             └── rand v0.9.2 (*)

advisories ok, bans FAILED, licenses ok, sources ok
```

c-git committed:
ci: update version of cargo shear to 1.11.2
